### PR TITLE
Fix CutOffFunctionTensorProduct::gradient.

### PR DESCRIPTION
**Pull request by luca-heltai**
This has been broken for a long time...

### CODE REVIEW

**Comment by bangerth** [comment]
Changed your mind?

**Comment by luca-heltai** [comment]
No. Didn't solve the issue. I figured that the test was bogus (always printing OK). I'm trying to figure out where the issue is, but it is less trivial than I thought...

File: tests/base/function_cutoff_03.cc [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    {
+      integral += values[i] * quad.weight(i);
+      grad_integral += gradients[i] * gradients[i] * quad.weight(i);
+    }
+  if (std::abs(integral - 1.0) > 1e-10)
+    deallog << "Value NOT OK" << std::endl;
+  else
+    deallog << "Value OK" << std::endl;
+  if (std::abs(grad_integral - exact[dim - 1]) > 1e-10)
+    deallog << "Gradient OK" << std::endl;
```
Comment (luca-heltai) [inline tests/base/function_cutoff_03.cc:72]: This should say "Gradient NOT OK", and even with the current changes, the gradient is not ok.

**Pull request closed without merging**